### PR TITLE
feat: fall back to NetEase Cloud Music when LRCLIB has no lyrics

### LINKED ISSUE #713: Lyrics not shown for NetEase Cloud Music (网易云音乐) on macOS
## Environment

- macOS 26.5.2
- Atoll 2.3.3
- NetEase Cloud Music (网易云音乐) 3.1.8.3368 — Electron/CEF-based

## Problem

When playing music in NetEase Cloud Music on macOS, the Atoll notch shows album artwork, track title and transport controls, but **no lyrics** are displayed, even with the lyrics feature enabled (`enableLyrics = 1`).

Interestingly, macOS's own Control Center "Now Playing" card **does** show lyrics for the exact same track (e.g. 杨坤 - 如果当时 (Live 版)). I assume this works because macOS can fall back to the Apple Music catalog to fill in lyrics, while Atoll only reads the `kMRMediaRemoteNowPlayingInfoLyrics` field via MediaRemote — and NetEase Cloud Music (an Electron app) does not publish lyrics metadata to MediaRemote.

## Expected behavior

It would be great if Atoll could show lyrics for NetEase Cloud Music as well. Possible approaches:

1. Fall back to an additional lyrics source when MediaRemote has no lyrics field (e.g. fetch lyrics by track title/artist from NetEase's public API).
2. At minimum, document in the README that lyrics only work with players that publish lyrics to MediaRemote (Apple Music, Spotify, YouTube Music).

## Notes

- Media controls (play/pause/next/seek, artwork, progress) all work fine with NetEase Cloud Music — only lyrics are missing.
- DockDoor has the exact same limitation (their README states lyrics support is for Spotify & Apple Music only), so this appears to be a common gap for Chinese music apps that don't use native macOS media APIs fully.

Thanks for the great app! The notch is genuinely useful.

### PR DESCRIPTION
**Pull request by MoFeng2223**
Closes #713

**Problem.** Atoll only fetches lyrics from LRCLIB. Most of NetEase Cloud Music's catalogue was never submitted there, so NetEase users see "No lyrics found" for the bulk of what they play.

**Change.** When LRCLIB has no lyrics for the playing track, Atoll now fetches them from NetEase Cloud Music's own lyrics API. LRCLIB is still asked first and its path is unchanged.

## How

- NetEase is asked when LRCLIB returns nothing, or returns rows that all fail its title/artist check (the branch that currently shows the raw `[{"id":…` JSON in the notch). If NetEase has nothing either, behaviour is as before.
- The player gives no song id, so the track is found again by searching title + artist and keeping the candidate that matches on title (no unrequested remix/karaoke/instrumental markers, reusing `LyricsSearchResults.carriesUnrequestedVersion`), lists an artist from the player's artist string, and is within 2 s of the playing duration.
- The LRC is parsed by the existing `LRCParser` and shown as served.

## Endpoints

- `GET https://music.163.com/api/search/get/web?s=<title artist>&type=1&limit=10`
- `GET https://music.163.com/api/song/lyric?id=<id>&lv=-1`

Not an official public API, but the endpoints NetEase's own web player has used for years, unauthenticated and stable across client releases. If they ever change, the result is "No lyrics found", same as today. Only the track title and artist are sent.

## Testing

`NetEaseLyricsTests` covers parsing, request building and candidate selection. Verified in the notch with NetEase Cloud Music as the source for all three paths: LRCLIB empty, LRCLIB unmatched, LRCLIB found.

## Summary by CodeRabbit

* **New Features**
  * Added a lyrics fallback to NetEase Cloud Music when LRCLIB has no matching lyrics.
  * Matches tracks by title, artist, and duration before displaying synchronized lyrics.
  * Antigravity cards now show Today/Week token usage and estimated API-equivalent costs, including cache-aware pricing.

* **Bug Fixes**
  * Improved handling of unmatched LRCLIB search responses.
  * Fixed Claude reset countdowns, usage-tab sizing, card title wrapping, language-server discovery, and pool dollar formatting.
  * Improved model pricing and usage recognition across supported providers.

* **Tests**
  * Added coverage for lyric matching, response parsing, request construction, and edge cases.

### CODE REVIEW

File: DynamicIsland/managers/NetEaseLyrics.swift [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+        else { return nil }
+        return URL(string: "https://music.163.com/api/search/get/web?s=\(encoded)&type=1&limit=10")
+    }
+
+    static func lyricURL(songID: Int) -> URL? {
+        // lv, kv and tv name the line-synced, karaoke and translated versions
+        // wanted; -1 is "whatever is current". A positive version number is a
+        // comparison against that version, and for some tracks it comes back
+        // with the body left empty. Only the LRC is shown.
+        URL(string: "https://music.163.com/api/song/lyric?id=\(songID)&lv=-1&kv=-1&tv=-1")
```
[the remark this answers is not in the document]
Comment (MoFeng2223) [reply DynamicIsland/managers/NetEaseLyrics.swift]: Checked against the endpoint rather than the docs: `tv=0` still returns `tlyric` (the value is treated as a version the client already holds, so anything older than the current translation gets the body). Leaving `tv` and `kv` out is what actually drops the translated and karaoke variants from the response, so the request is now `lv=-1` only, with a test asserting the query is exactly `id` + `lv` (a534bba).

**Comment by MoFeng2223** [comment]
On the cache-identity note (`LyricsLookupKey` without duration): that key is the existing one, and it deliberately collapses `Song` and `Song (Live)` — `normalizedLyricsTitle` strips `live`/`remaster` before the key is built, and `versionMarkers` excludes them for the same reason, so LRCLIB results are already cached that way today. This PR keeps the same identity so the NetEase path behaves like the LRCLIB one; changing what the cache considers the same track would affect both sources and is out of scope here.

**Review by StudioKeys — APPROVED** [review]
